### PR TITLE
update Identity version

### PR DESCRIPTION
**Pull request by g2vinay**
Updates Identity version to 1.0.4
Updates Identity core dep to 1.3.0

### CODE REVIEW

File: sdk/identity/azure-identity/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Release History
 
+## 1.0.4 (2020-03-10)
+- Upgrade `azure-core` dependency to 1.0.4
```
Comment (alzimmermsft) [inline sdk/identity/azure-identity/CHANGELOG.md]:
```suggestion
- Upgraded `azure-core` dependency to 1.0.4.
```

File: sdk/identity/azure-identity/pom.xml [hunk 2]
```diff
@@ -19,11 +19,34 @@
...
+              <code>java.method.added</code>
+              <class>com.azure.core.http.ProxyOptions</class>
+              <justification>The new API added in azure core can be ignored, as it doesn't break the API. </justification>
+            </item>
+          </revapi.ignore>
+        </analysisConfiguration>
+        </configuration>
+      </plugin>
+    </plugins>
+  </build>
```
Comment (alzimmermsft) [inline sdk/identity/azure-identity/pom.xml]: Could this configuration be added into the revapi configuration JSON file?

Comment (g2vinay) [reply sdk/identity/azure-identity/pom.xml]: we'll need to make a separate file for that, which @JonathanGiles isn't covinvced about yet.
Putting it in the same JSON file will apply it to parent level config, and that will get applied to azure-core itself, i.e we will stop checking azure-core changes itself on azure-core.

We'll be removing this config from pom file, after release for now.

File: sdk/identity/azure-identity/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # Release History
 
+## 1.0.4 (2020-03-10)
+- Upgraded `azure-core` dependency to 1.0.4
+
+
 ## 1.0.3 (2020-01-013)
```
Comment (mssfang) [inline sdk/identity/azure-identity/CHANGELOG.md]: NIT: 2020-01-13

**Pull request merged**